Main.hs

```diff
@@ -11,28 +11,32 @@ import qualified Data.List as L
 import Text.Printf (printf)
 
 multByIndex :: [Int] -> [Int]
-multByIndex = undefined
+multByIndex xs = zipWith (*) xs [0..]
 
 powerByIndex :: [Int] -> [Int]
-powerByIndex = undefined
+powerByIndex xs = zipWith (^) xs [0..]
 
 productOfDifference :: [Int] -> Int
-productOfDifference = undefined
+productOfDifference xs
+  | length (xs) < 2 = error "List is too short"
+  | otherwise = product $ zipWith (-) xs (drop 1 xs)
 
-isSorted :: [Int] -> Bool 
-isSorted xs = undefined
+isSorted :: [Int] -> Bool
+isSorted xs = and $ zipWith (<=) xs (drop 1 xs)
```

Comment:
Instead of `drop 1 xs` you could use `tail xs`. It's not a problem here, just an advise

Reply:
I have tried it before the drop 1 xs but I got a warning: 
Main_second.hs:22:43: warning: [GHC-63394] [-Wx-partial]
    In the use of ‘tail’
    (imported from Prelude, but defined in GHC.List):
    "This is a partial function, it throws an error on empty lists. Replace it with drop 1, or use pattern matching or Data.List.uncons instead. Consider refactoring to use Data.List.NonEmpty."
   |
22 |   | otherwise = product $ zipWith (-) xs (tail xs)
   |                                           ^^^^
Ok, one module loaded.

Reply:
@YassineFnn This warning is indeed something you shouldn't ignore. 
`drop 1 xs` cannot fail on an empty list indeed, but it's because it's defined to be empty on an empty list. What I would do is to pattern match on the list, considering the empty and non-empty cases.